walletlibrary/src/main/java/com/microsoft/walletlibrary/requests/handlers/OpenId4VCIRequestHandler.kt

```diff
@@ -71,7 +67,99 @@ internal class OpenId4VCIRequestHandler(private val libraryConfiguration: Librar
                     it as Exception
                 )
             }
-        TODO("Map data models and finally return VerifiedIdRequest.")
+        throw OpenId4VciValidationException(
+            "Failed to validate or transform credential metadata",
+            VerifiedIdExceptions.MALFORMED_CREDENTIAL_METADATA_EXCEPTION.value
+        )
+    }
+
+    private fun decodeCredentialOffer(rawRequest: Any): CredentialOffer {
+        try {
+            // Deserialize the raw request to a CredentialOffer object.
+            return libraryConfiguration.serializer.decodeFromString(
+                CredentialOffer.serializer(),
+                rawRequest as String
+            )
+        } catch (exception: Exception) {
+            throw OpenId4VciValidationException(
+                "Failed to decode CredentialOffer ${exception.message}",
+                VerifiedIdExceptions.MALFORMED_CREDENTIAL_OFFER_EXCEPTION.value,
+                exception
+            )
+        }
+    }
+
+    private fun getSupportedCredentialConfigurationId(
+        credentialMetadata: CredentialMetadata,
+        credentialOffer: CredentialOffer
+    ): CredentialConfiguration {
+        // Get only the supported credential configuration ids from the credential metadata from the list in credential offer.
+        val configIds = credentialOffer.credential_configuration_ids
+        val supportedCredentialConfigurationIds =
+            credentialMetadata.getSupportedCredentialConfigurations(configIds)
+        if (supportedCredentialConfigurationIds.isEmpty())
+            throw OpenId4VciValidationException(
+                "Request does not contain supported credential configuration.",
+                VerifiedIdExceptions.MALFORMED_CREDENTIAL_METADATA_EXCEPTION.value
+            )
+        return supportedCredentialConfigurationIds.first()
```

Comment:
Is there a case when we would need to support more than one?

Reply:
May be when we support bulk issuance or nested issuance.